src/lib/storage/lz4_segment.cpp

```diff
@@ -0,0 +1,147 @@
+#include "lz4_segment.hpp"
+
+#include "resolve_type.hpp"
+#include "storage/vector_compression/base_compressed_vector.hpp"
+#include "utils/assert.hpp"
+#include "utils/performance_warning.hpp"
+
+#include "lib/lz4.h"
+
+namespace opossum {
+
+template <typename T>
+LZ4Segment<T>::LZ4Segment(const std::shared_ptr<const pmr_vector<char>>& compressed_data,
+                          const std::shared_ptr<const pmr_vector<bool>>& null_values,
+                          const std::shared_ptr<const pmr_vector<size_t>>& offsets, const int compressed_size,
+                          const int decompressed_size, const size_t num_elements)
+    : BaseEncodedSegment{data_type_from_type<T>()},
+      _compressed_data{compressed_data},
+      _null_values{null_values},
+      _offsets{offsets},
+      _compressed_size{compressed_size},
+      _decompressed_size{decompressed_size},
+      _num_elements{num_elements} {}
+
+template <typename T>
+std::shared_ptr<const pmr_vector<char>> LZ4Segment<T>::compressed_data() const {
```

Comment:
Are these Java'esk accessors necessary?

Reply:
Probably not. We'll remove them.

Reply:
Needed or used?

Reply:
The null value accessor is needed in the iterators and the offset accessor is needed for testing.

Reply:
Ack.